oak/server/BUILD

```diff
@@ -132,19 +132,41 @@ cc_library(
     ],
 )
 
+# Wrapper rule to expose the resulting static library as a statically linked cc_library and
+# corresponding header so that it can be depended on by other cc_library and cc_binary rules.
+#
+# TODO: There seems to be something wrong with this rule related to caching in Bazel.
+# To reproduce:
+# - change src/lib.rs and introduce a syntax error
+# - bazel build //rust/oak_runtime:wrapper
+# - this should produce a compile error, instead Bazel is still caching the old artifact
+# The final link of //oak/server/asylo:oak uses oak_runtime_stripped.a not libwrapper.a. Similarly,
+# a change to the Rust file does induce a rebuild when doing bazel build //oak/server/dev:oak. So is
+# Bazel maybe doing something special for a target with a single src that is itself a .a file?
+cc_library(
+    name = "rust_wrapper",
+    srcs = ["//sdk/rust/oak_runtime_compat:oak_runtime_stripped"],
+    hdrs = ["oak_runtime.h"],
+    linkstatic = True,
+    visibility = ["//oak/server:__subpackages__"],
+)
+
 cc_library(
     name = "oak_runtime",
     srcs = ["oak_runtime.cc"],
-    hdrs = ["oak_runtime.h"],
+    hdrs = [
+        "oak_runtime.h",
```

Comment:
Bazel rules are usually non-overlapping, I'm surprised this actually works in fact. What's the reasoning for moving the wrapper here but leaving the wrapped code behind? It seems to break an otherwise encapsulated `cc_library`.

Reply:
The `oak_runtime` rule here is correct, its the existing cpp runtime. In the `rust_wrapper` rule I had forgotten to rename `oak_runtime.h`->`rust_check.h` but as the header is not actually used it compiled fine. I think this has caused some confusion- all the rust code has moved, and all the existing cpp runtime code has remained plus the (now named `rust_check.h`) header file for compiling in the rust compat module to check it works. Does that make sense?

Reply:
Ah I understand you now. I saw the `rust_check.h` as an artefact of the c code to phase out, but you're right its implementation is in the rust impl so I will move it there